File: csharp/src/Behavioral/Observer/DvdSubscriber.cs
namespace Behavioral.Observer {


//DvdSubscriber.java - the observer

public class DvdSubscriber {  
   private string subscriberName; 
   
   public DvdSubscriber(string subscriberNameIn) {
       this.subscriberName = subscriberNameIn;
   }
   
   public string getSubscriberName() {
       return this.subscriberName;
   }
   
   public string newDvdRelease(DvdRelease newDvdRelease, 
                             string subscriptionListName) {
       return "Hello " + this.getSubscriberName() + 
                          ", subscriber to the " + 
                             subscriptionListName + 
                          " DVD release list.\n" +
			"The new Dvd " + 
                            newDvdRelease.getDvdName() + 
                          " will be released on " + 
                            newDvdRelease.getDvdReleaseMonth() + "/" + 
                            newDvdRelease.getDvdReleaseDay() + "/" + 
                            newDvdRelease.getDvdReleaseYear() + ".";
   }
   
   public string updateDvdRelease(DvdRelease newDvdRelease, 
                                string subscriptionListName) {
       return "Hello " + this.getSubscriberName() + 
                          ", subscriber to the " + 
                            subscriptionListName +
                          " DVD release list.\n" +
         "The following DVDs release has been revised: " + 
         newDvdRelease.getDvdName() + " will be released on " + 
         newDvdRelease.getDvdReleaseMonth() + "/" + 
         newDvdRelease.getDvdReleaseDay() + "/" + 
         newDvdRelease.getDvdReleaseYear() + ".";
   }   
}

}

File: scala/src/main/scala/Behavioral/Observer/DvdSubscriber.scala
package Behavioral.Observer;


//DvdSubscriber - the observer

import java.util.*;
import org.apache.commons.lang3.StringUtils;

public class DvdSubscriber {  
   private String subscriberName; 
   
   public DvdSubscriber(String subscriberNameIn) {
       this.subscriberName = subscriberNameIn;
   }
   
   public String getSubscriberName() {
       return this.subscriberName;
   }
   
   public String newDvdRelease(DvdRelease newDvdRelease, 
                             String subscriptionListName) 
	{
			List<String> retval = new ArrayList<String>();
       retval.add("Hello " + this.getSubscriberName() + 
                          ", subscriber to the " + 
                             subscriptionListName + 
                          " DVD release list.");
       retval.add("The new Dvd " + 
                            newDvdRelease.getDvdName() + 
                          " will be released on " + 
                            newDvdRelease.getDvdReleaseMonth() + "/" + 
                            newDvdRelease.getDvdReleaseDay() + "/" + 
                            newDvdRelease.getDvdReleaseYear() + ".");
		 return StringUtils.join(retval,"\n");
   }
   
   public String updateDvdRelease(DvdRelease newDvdRelease, String subscriptionListName) 
	{
			List<String> retval = new ArrayList<String>();
       retval.add("Hello " + this.getSubscriberName() + 
                          ", subscriber to the " + 
                            subscriptionListName +
                          " DVD release list.");
       retval.add(
         "The following DVDs release has been revised: " + 
         newDvdRelease.getDvdName() + " will be released on " + 
         newDvdRelease.getDvdReleaseMonth() + "/" + 
         newDvdRelease.getDvdReleaseDay() + "/" + 
         newDvdRelease.getDvdReleaseYear() + ".");
		 return StringUtils.join(retval,"\n");
   }   
}
